src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+| `COCKROACH_LOG_MAX_SYNC_DURATION`            | The maximum duration the file sink is allowed to take to write a log entry before we fatal the process for a disk stall. | `20s`   | `40s`                                       |
+
+### To enable WAL Failover
+
+[XXX](): YOU ARE HERE
+
+Setup your cluster for WAL failover with either [multiple stores](#multi-store-config) or a [side disk](#single-store-config).
+
+For [multiple stores](#multi-store-config), pass `--wal-failover=among-stores` to [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}).
+
+For a [side disk on a single store config](#single-store-config), pass `--wal-failover={ path-to-side-drive-directory }` to [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}).
+
+Use remote log sinks, or if you use file-based logging, enable asynchronous buffering of `file-groups` log sinks:
+
+{% include {{ page.version.version }}/misc/logging-flags.md %}
+
+Change the value of `COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT` by setting it as follows:
+
+1. Before starting the cluster, set it using the [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables) environment variable.
+2. After starting the cluster, set it by changing the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting.
+
+When WAL failover is enabled, we recommended using `40s` for this setting. If WAL failover is not enabled, **do not change the default value of this setting**.
+
+### To monitor WAL Failover
+
+{% include {{ page.version.version }}/wal-failover-metrics.md %}
+
+### WAL Failover behavior
+
+If a disk stalls for less than the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables), WAL failover will trigger and the node will continue to operate normally.
+
+If a disk stalls for longer than the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables), a WAL failover will trigger. Following that, since the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables) has been exceeded, the node will go down.
+
+In a [multi-store](#multi-store-config) cluster, if a disk for a store has a transient stall, WAL will failover to the second store's disk. When the stall on the first disk clears, the WAL will failback to the first disk. WAL failover will daisy-chain from store _A_ to store _B_ to store _C_.
+
+However, failback will not cascade until all drives become available again: that is, if the disk for store _A_ unstalls while the disk for store _B_ is still stalled, store _C_ will not failback to store _A_ until the intermediate disk _B_ also becomes available again.
```

Comment:
I'm not sure what this sentence is saying, but I don't think it's correct. We should remove it.

Reply:
deleted!